mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -145,7 +145,7 @@ def SYCL_AccessorType
                         "unsigned":$dimension,
                         "mlir::sycl::MemoryAccessMode":$accessMode,
                         "mlir::sycl::MemoryTargetMode":$targetMode,
-                        "llvm::SmallVector<mlir::Type, 4>":$body);
+                        ArrayRefParameter<"mlir::Type">:$body);
```

Comment:
This only changes the interface: the storage type will be `SmallVector<Type>`, whereas the input/output arguments will be `ArrayRef<Type>`.